### PR TITLE
No longer in-place modify input to apply

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Update based lightning PR#[474](https://github.com/PennyLaneAI/pennylane-lightning/pull/474). 
**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Hi @multiphaseCFD, thank you for propagating this important update to LGPU.

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 1]
```diff
@@ -252,6 +252,8 @@ def __init__(
 
             super().__init__(wires, shots=shots, r_dtype=r_dtype, c_dtype=c_dtype)
 
+            self._dp = DevPool()
```
Comment (mlxd) [inline pennylane_lightning_gpu/lightning_gpu.py:255]: Why was this moved? Is it needed for this PR?

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 2]
```diff
@@ -557,10 +558,12 @@ def apply(self, operations, **kwargs):
                     self._apply_state_vector_GPU(
                         operations[0].parameters[0].copy(), operations[0].wires
                     )
-                    del operations[0]
+                    operations = operations[1:]
+                    # del operations[0]
```
Comment (mlxd) [inline pennylane_lightning_gpu/lightning_gpu.py:562]: Can this comment be removed?

File: pennylane_lightning_gpu/lightning_gpu.py [hunk 3]
```diff
@@ -557,10 +558,12 @@ def apply(self, operations, **kwargs):
...
                 elif isinstance(operations[0], BasisState):
                     self._apply_basis_state_GPU(operations[0].parameters[0], operations[0].wires)
-                    del operations[0]
+                    operations = operations[1:]
+                    # del operations[0]
```
Comment (mlxd) [inline pennylane_lightning_gpu/lightning_gpu.py:566]: Same here

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -5,14 +5,17 @@
  * Add Sparse Hamiltonian support for expectation value calculation.
  [(#127)] (https://github.com/PennyLaneAI/pennylane-lightning-gpu/pull/127)
 
-
+ * `apply` no longer mutates the inputted list of operations.
```
Comment (vincentmr) [inline .github/CHANGELOG.md:8]: I think this might go in `Bug fixes`.

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 2 above)
Comment (vincentmr) [inline pennylane_lightning_gpu/lightning_gpu.py:562]: Delete comment.

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 3 above)
Comment (vincentmr) [inline pennylane_lightning_gpu/lightning_gpu.py:566]: Delete comment.

**Review by vincentmr — COMMENTED** [review]
Few minor things, but looking good. Thanks @multiphaseCFD

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning_gpu/lightning_gpu.py:255]: This line was moved here because `self_.dp` is needed for both single GPU backend and multi-node/multi-GPU backend, but was only kept at the multi-node version in the previous implementation. It was a bug that didn't show up in previous unit tests check. We need this change to pass the CI checks.

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 3 above)
Comment (multiphaseCFD) [reply pennylane_lightning_gpu/lightning_gpu.py]: Done!

Comment (multiphaseCFD) [reply pennylane_lightning_gpu/lightning_gpu.py]: Done!

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 2 above)
Comment (multiphaseCFD) [reply pennylane_lightning_gpu/lightning_gpu.py]: Done!

File: .github/CHANGELOG.md (hunk 4 above)
Comment (multiphaseCFD) [reply .github/CHANGELOG.md]: Updated!

File: pennylane_lightning_gpu/lightning_gpu.py (hunk 1 above)
Comment (mlxd) [reply pennylane_lightning_gpu/lightning_gpu.py:255]: I think I recall we moved this in the same way in the past to ensure it is single and multinode accessible. Maybe another merge overwrite this change. All good to keep it here.

**Review by mlxd — APPROVED** [review]
Approved, pending CI success

**Review by vincentmr — APPROVED** [review]
Looks good!
